duckduckgo/duckduckgo-autofill
# Email Protection sign-up tooltip

**Reviewer:** @GioSensation @shakyShane 
**Asana:** https://app.asana.com/0/0/1203490723369675/f

## Description
**Current state** -- When autofill instantiates the device interface, [it sets up the UI Controller in the constructor](https://github.com/duckduckgo/duckduckgo-autofill/blob/b16d363471003b39973b0f5418d3af2d37e657ae/src/DeviceInterface/InterfacePrototype.js#L76)

As part of this work, in the extensions, we want to show Email Signup tooltip when the user doesn't have Email Protection enabled, and Email Protection autofill when they do. This requires moving the UI Controller set up to after `setupAutofill` has completed so we know if Email Protection is enabled or not.

## Steps to test
1. Use this branch, changing the `emailProtection_incontext_signup` value in `src/Settings.js` to `true`
2. Build and use with an unpacked extension (see development steps)
3. With Email Protection disabled, go to https://spreadprivacy.com/privacy-weekly-newsletter/
4. Confirm the tootip shows with the Email Protection notice
5. Enable email protection
6. With Email Protection enabled, go to https://spreadprivacy.com/privacy-weekly-newsletter/
7. Confirm the tootip now shows with the Email Protection autofill
